Verify manifest before accepting

miminar opened the pull request:
Referenced blobs need to be checked for presence in a destination repository before accepting incoming manifest. Otherwise, user will be able to pull image without previous access to its blobs.

Resolves [bz#1388018](https://bugzilla.redhat.com/show_bug.cgi?id=1388018)

Blocked on #11925

miminar commented:
/cc @legionus, @mfojtik

miminar commented:
[test]

miminar commented:
Yum flake #8571. re-[test]

kargakis commented on pkg/dockerregistry/server/manifestschema1handler.go:
```diff
@@ -0,0 +1,173 @@
+		if err != nil {
+			if err != distribution.ErrBlobUnknown {
+				errs = append(errs, err)
+			}
```
You probably want to continue here otherwise you will append the error below too, right?

miminar replied:
Good catch!

kargakis replied:
Same for the schema2 handler

legionus commented:
@miminar Depends on what PR (this or https://github.com/openshift/origin/pull/11925) would be merged first we get a conflict. In the https://github.com/openshift/origin/pull/11925 I have changed your `Registry: moved manifest schema operations to new files` commit.

miminar commented:
> Depends on what PR (this or #11925) would be merged first we get a conflict. In the #11925 I have changed your Registry: moved manifest schema operations to new files commit.

@legionus Let's go with yours first.

miminar commented:
the yum flake beast went rampage

miminar commented:
[test]

miminar commented:
Yum flake #8571. re-[test]

miminar commented:
Updated 1st commit to match the one in #11925.

soltysh commented:
LGTM, tagging for merge after green tests.

miminar commented:
@mfojtik, @pweil-, @soltysh someone please merge.

soltysh commented:
[merge]
